modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -59,6 +66,570 @@ static bool s_first = true;
 static int s_shards = 0;
 static std::mutex s_lmdbStartupLock;
 
+std::pair<uint32_t, uint32_t> LMDBBackend::getSchemaVersionAndShards(std::string& filename)
+{
+  // cerr << "getting schema version for path " << filename << endl;
+
+  uint32_t schemaversion;
+
+  int rc;
+  MDB_env* env = nullptr;
+  MDB_dbi dbi;
+  MDB_val key, data;
+  MDB_txn* txn = nullptr;
+
+  if ((rc = mdb_env_create(&env)) != 0) {
+    throw std::runtime_error("mdb_env_create failed");
+  }
+
+  if ((rc = mdb_env_set_maxdbs(env, 20)) != 0) {
+    mdb_env_close(env);
+    throw std::runtime_error("mdb_env_set_maxdbs failed");
+  }
+
+  if ((rc = mdb_env_open(env, filename.c_str(), MDB_NOSUBDIR, 0600)) != 0) {
```

Comment:
I suspect that this will also create the LMDB if it does not exist yet?

As the mapsize has not been explicitly set with `mdb_env_set_mapsize`, it will default to 10 MB. A value of 0 would probably prevent this. The mapsize is always automatically adjusted up to a bigger mapsize set in the LMDB.